packages/client/src/objectSet/ObjectSetListenerWebsocket.test.ts

```diff
@@ -226,7 +252,59 @@ describe("ObjectSetListenerWebsocket", async () => {
 
       describe("successfully subscribed", () => {
         beforeEach(() => {
+          console.log("ID WAS", currentSubscriptionId);
           respondSuccessToSubscribe(ws, subReq1);
+          console.log("ID IS NOW", currentSubscriptionId);
+          console.log("responded to subscribe", subReq1.id);
+        });
+
+        it("should correctly return regular updates", async () => {
+          const idNum2 = currentSubscriptionId;
+          console.log("idNum2", idNum2);
+
+          sendObjectUpdateResponse(ws, `${idNum2}`);
+          await listenerPromise;
+          expect(listener.onChange).toHaveBeenCalled();
+          expect(updateReceived).toMatchInlineSnapshot(`
+            {
+              "object": {
+                "$apiName": "Employee",
+                "$objectType": "Employee",
+                "employeeId": 1,
+              },
+              "state": "ADDED_OR_UPDATED",
+            }
+          `);
+        });
+
+        it("should correctly return reference update", async () => {
+          const idNum2 = currentSubscriptionId;
+          console.log("idNum2", idNum2);
+
+          sendReferenceUpdatesResponse(ws, `${idNum2}`);
+          await listenerPromise;
+          expect(listener.onChange).toHaveBeenCalled();
+          expect(updateReceived).toMatchInlineSnapshot(`
+            {
+              "object": {
+                "$apiName": "Employee",
+                "$objectType": "Employee",
+                "$primaryKey": {
```

Comment:
This is in a weird state right now. The type of the primary key on a reference update is a Record of property type to property value instead of just a value. Since I haven't actually been able to hit the backend, I haven't been able to confirm what this looks like. I suspect I just need to index in via the primary key type. This PR won't merge until I test and resolve this

Reply:
So the primary key looks funky but will be fixed when I can confirm what the response is